### PR TITLE
cmd/snap-repair: implement the repair run loop

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #3930

File: cmd/snap-repair/cmd_run_test.go [hunk 1]
```diff
@@ -20,14 +20,39 @@
...
 
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/sysdb"
 	repair "github.com/snapcore/snapd/cmd/snap-repair"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
 )
 
 func (r *repairSuite) TestRun(c *C) {
+	r1 := sysdb.InjectTrusted(r.storeSigning.Trusted)
```
Comment (mvo5) [inline cmd/snap-repair/cmd_run_test.go]: Thanks for this test!

File: cmd/snap-repair/cmd_run.go [hunk 2]
```diff
@@ -37,7 +43,60 @@ func init() {
...
+		panic(fmt.Sprintf("cannot setup base url: %v", err))
+	}
+}
+
 func (c *cmdRun) Execute(args []string) error {
-	fmt.Fprintf(Stdout, "run is not implemented yet\n")
+	if err := os.MkdirAll(dirs.SnapRunRepairDir, 0700); err != nil {
+		return err
+	}
+	flock, err := osutil.NewFileLock(filepath.Join(dirs.SnapRunRepairDir, "lock"))
```
Comment (mvo5) [inline cmd/snap-repair/cmd_run.go]: Should we test the lock file behaviour?

**Comment by mvo5** [comment]
There is one small test update required:
```
2017-09-19 22:42:47 Error executing linode:ubuntu-core-16-64:tests/main/snap-repair : 
...
+ /usr/lib/snapd/snap-repair run
+ MATCH 'run is not implemented yet'
error: pattern not found, got:
-----
```

File: cmd/snap-repair/cmd_run.go [hunk 3]
```diff
@@ -37,7 +43,60 @@ func init() {
...
+	if osutil.GetenvBool("SNAPPY_USE_STAGING_STORE") {
+		baseurl = "https://api.staging.snapcraft.io/v2/"
+	} else {
+		baseurl = "https://api.snapcraft.io/v2/"
+	}
+
+	var err error
+	baseURL, err = url.Parse(baseurl)
+	if err != nil {
+		panic(fmt.Sprintf("cannot setup base url: %v", err))
```
Comment (chipaca) [inline cmd/snap-repair/cmd_run.go]: you could do `logger.Panicf`` here

**Comment by mvo5** [comment]
This is merged via the merge of https://github.com/snapcore/snapd/pull/3985

**Pull request closed without merging**